.github/workflows/ci.yml

```diff
@@ -0,0 +1,564 @@
+name: Alfresco Community Repo CI
+
+on:
+  pull_request:
+    branches:
+      - feature/**
+      - fix/**
+  push:
+    branches:
+      - feature/**
+      - fix/**
+  workflow_call:
+  workflow_dispatch:
+
+env:
+  DOCKERHUB_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+  DOCKERHUB_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+  GITHUB_ACTIONS_DEPLOY_TIMEOUT: 60
+  LOG_WARN: "-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn"
+  MAVEN_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
+  MAVEN_USERNAME: ${{ secrets.NEXUS_USERNAME }}
+  QUAY_PASSWORD: ${{ secrets.QUAY_PASSWORD }}
+  QUAY_USERNAME: ${{ secrets.QUAY_USERNAME }}
+  TAS_ENVIRONMENT: ./packaging/tests/environment
+  TAS_SCRIPTS: ../alfresco-community-repo/packaging/tests/scripts
+
+jobs:
+  prepare:
+    name: "Prepare"
+    runs-on: ubuntu-latest
+    if: >
+      ! contains(github.event.head_commit.message, '[skip tests]') &&
+      ! contains(github.event.head_commit.message, '[force]')
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Alfresco/acs-community-packaging/.github/actions/get-build-info@feature/ACS-3844_migrate-to-GHA
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.25.0
+      - name: "Init"
+        run: bash ./scripts/ci/init.sh
+      - name: "Prepare environment"
+        run: bash ./scripts/ci/prepare.sh
+      - name: "Clean Maven cache"
+        if: ${{ always() }}
+        run: bash ./scripts/ci/cleanup_cache.sh
+
+  veracode:
+    name: "Source Clear Scan (SCA)"
+    runs-on: ubuntu-latest
+    needs: [prepare]
+    if: >
+      ((github.ref_name == 'master' || contains(github.ref_name, 'release/')) && github.event_name != 'pull_request') &&
+      ! contains(github.event.head_commit.message, '[skip tests]') &&
+      ! contains(github.event.head_commit.message, '[force]')
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Alfresco/acs-community-packaging/.github/actions/get-build-info@feature/ACS-3844_migrate-to-GHA
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: "Init"
+        run: bash ./scripts/ci/init.sh
+      - uses: Alfresco/alfresco-build-tools/.github/actions/veracode@master
+        with:
+          srcclr-api-token: ${{ secrets.SRCCLR_API_TOKEN }}
+      - name: "Clean Maven cache"
+        if: ${{ always() }}
```

Comment:
cache is never failed upon failure (see https://github.com/actions/cache/blob/main/action.yml#L24)

why you need this?

Reply:
We need `always()` to run the `Clean Maven cache`  regardless of the `veracode` step result.

Reply:
I mean, if the reason is that you want to prevent the cache from being saved without the cleanup script has run, `always` is not really necessary because cache **never** get saved if the job is failing (because a failed job 99% has an incomplete cache)

Reply:
I got it. I removed `always()` from all `Maven cache clean` steps.